src/main/java/ro/secur/auth/service/UserService.java

```diff
@@ -88,8 +103,58 @@ public void changePassword(String username, ChangePasswordRequest request) {
         }
     }
 
+    public void forgotPassword(String email, HttpServletRequest request) {
+
+        String email1 = email.replace("{\"email\":{\"email\":\"", "");
+        String email2 = email1.replace("\"}}", "");
+        UserInfoEntity userInfoEntity = userInfoRepository.findByEmail(email2);
+        UserEntity userEntity = userRepository.findByUserInfoEntity(userInfoEntity);
+
+        if (userEntity != null) {
+            userEntity.setResetToken(UUID.randomUUID().toString());
+
+            save(userEntity);
+
+            String appUrl = request.getScheme() + "://" + request.getServerName() + ":" + request.getServerPort();
+
+            String token = userEntity.getResetToken();
+
```

Comment:
Cred ca ai putea imparti toata metoda asta .  Vad ca aici faci si replaceul dai save in db si apoi mai creezi si alt obiect de mail. (Fiecare metoda ar trebui facuta cant mai simplu si fiecare cu functionalitatea ei ). 
Vad ca mai iei de pe request si pui direct in entitate ceea ce nu cred ca e corect. Ar trebuie sa treci si prin layerul de DTO.